snapcraft/__init__.py

```diff
@@ -183,23 +159,12 @@ def __init__(self, name, options):
 
     # The API
     def pull(self):
-        """Pull the source code and/or internal prereqs to build the part.
-
-        By default, the base implementation for pull will use the following
-        part properties to retrieve source code:
-
-        - source
-        - source-branch
-        - source-tag
-        - source-type
-
-        If source is empty or does not exist, the phase will be skipped.
+        """Pull any code and/or internal prereqs to build the part.
 
         Override or inherit from this method if you need to implement or
         enhance with custom pull logic.
         """
-        if getattr(self.options, 'source', None):
-            sources.get(self.sourcedir, self.builddir, self.options)
+        pass
```

Comment:
So the idea is that the lifecycle handles sources instead of the plugin. That makes source handling a higher level concept that all plugins can use and don't have implement explicitly. They can just assume there is a source directory and that was magically populated for them with what ever the source directives specify.